src/lib/interruptible/interruptible.ml

```diff
@@ -57,6 +57,12 @@ module T = struct
   let return a =
     {interruption_signal= Deferred.never (); d= Deferred.Result.return a}
 
+  module Can_err = struct
+    type (_, 'a, 's) t =
+      | () : (('a, 's) Result.t, 'a, 's) t
```

Comment:
woah cool, didn't realize you could use unit as a constructor.

Reply:
This does seem cool but also unused

Reply:
thanks @cmr

Reply:
Coolness aside, I worry about overloading common syntax, might be confusing when reading code.

Reply:
I deleted this because it was unused